packages/myst-jb-toc/src/toc.ts

```diff
@@ -0,0 +1,48 @@
+import yaml from 'js-yaml';
+
+export const TOC_FORMAT = 'jb-book';
+export const TOC_FORMAT_ARTICLE = 'jb-article';
+
+export type TocOptions = {
+  path?: string;
+  filename?: string;
+  ci?: boolean;
+};
+
+export type JupyterBookPart = {
+  caption?: string;
+  chapters?: JupyterBookChapter[];
+};
+
+export type JupyterBookChapter = {
+  file?: string;
+  url?: string;
+  title?: string;
+  glob?: string;
+  sections?: JupyterBookChapter[];
+};
+
+export type TOC = {
+  format: string;
+  root: string;
+  sections?: JupyterBookChapter[];
+  chapters?: JupyterBookChapter[];
+  parts?: JupyterBookPart[];
+};
+
+export function parseTOC(contents: string): TOC {
+  const toc = yaml.load(contents) as any;
```

Comment:
Maybe worth a specific error return here - passing on the yaml validation error if it exists, but giving some context to know that the yaml failed reading a toc?

Reply:
I'm not sure what the pattern for re-throwing an exception is for TS. Can you shed any light?

Reply:
`throw new Error('Could not load TOC yaml')`

I usually print the initial error message (err.stack or trace?) to a debug log as well.